### PR TITLE
Name changes, subgroup check

### PR DESCRIPTION
**Pull request by martstr**

### CODE REVIEW

File: src/Shared/AnonymousTokensShared/Protocol/Initiator.cs [hunk 1]
```diff
@@ -85,18 +85,14 @@ public bool VerifyProof(X9ECParameters ecParameters, ECPoint P, ECPoint Q, BigIn
         /// <param name="z">Response from the Chaum-Pedersen proof</param>
         /// <param name="r">Masking of the initial point</param>
         /// <returns>A randomised signature W on the point chosen by the initiator</returns>
-        public ECPoint RandomiseToken(X9ECParameters ecParameters, ECPoint P, ECPoint Q, BigInteger c, BigInteger z, BigInteger r)
+        public ECPoint ArmToken(X9ECParameters ecParameters, ECPoint P, ECPoint Q, BigInteger c, BigInteger z, BigInteger r)
```
Comment (HenrikWM) [inline src/Shared/AnonymousTokensShared/Protocol/Initiator.cs:88]: Just curious about "arm" here. Is it an abbreviation or does it mean to "arm" the token (as in "arming the torpedoes")?

**Review by HenrikWM — COMMENTED** [review]
Just had a quick question :)

File: src/Shared/AnonymousTokensShared/Protocol/Initiator.cs (hunk 1 above)
Comment (martstr) [reply src/Shared/AnonymousTokensShared/Protocol/Initiator.cs:88]: As in the latter. Not going to insist.

Comment (HenrikWM) [reply src/Shared/AnonymousTokensShared/Protocol/Initiator.cs:88]: Is it a common phrase in cryptography? Or can we assess some synonyms which might also work? Does the comment "It first verifies that the incoming token is well-formed, and then removes the previously applied mask." help us with alternative ideas for a name? `RemoveMask` is perhaps one alternative, but then I'd prefer `RandomizeToken` (if the method actually is randomising something) :)

**Comment by HenrikWM** [comment]
Looks good!

**Pull request merged**